[CI] XPackRestIT test {p0=transform/transforms_stats_continuous/Test get continuous transform stats} failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/zjolr4cedf4na/tests/:x-pack:plugin:yamlRestTest/org.elasticsearch.xpack.test.rest.XPackRestIT/test%20%7Bp0=transform%2Ftransforms_stats_continuous%2FTest%20get%20continuous%20transform%20stats%7D

**Reproduction line:**
`./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=transform/transforms_stats_continuous/Test get continuous transform stats}" -Dtests.seed=B9BA0605278FAFC6 -Dtests.locale=it-IT -Dtests.timezone=Asia/Damascus -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.test.rest.XPackRestIT&tests.test=test%20%7Bp0%3Dtransform/transforms_stats_continuous/Test%20get%20continuous%20transform%20stats%7D

**Failure excerpt:**
```
java.lang.AssertionError: Failure at [transform/transforms_stats_continuous:35]: expected [2xx] status code but api [transform.stop_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Failed to update transform task [airline-transform-stats-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-stats-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:320)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:318)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:295)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:275)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:267)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1337)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1422)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1402)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:132)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:112)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:710)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"status_exception","reason":"Failed to update transform task [airline-transform-stats-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-stats-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:320)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:318)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:295)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:275)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:267)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1337)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1422)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1402)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:132)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:112)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:710)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":409}]

  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:510)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:484)
  at jdk.internal.reflect.GeneratedMethodAccessor14.invoke(null:-1)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.lang.AssertionError: expected [2xx] status code but api [transform.stop_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Failed to update transform task [airline-transform-stats-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-stats-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:320)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:318)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:295)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:275)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:267)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1337)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1422)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1402)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:132)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:112)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:710)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"status_exception","reason":"Failed to update transform task [airline-transform-stats-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-stats-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:320)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:299)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:318)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:295)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:275)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:267)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1337)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1422)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1402)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:132)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:112)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:710)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":409}]

    at org.junit.Assert.fail(Assert.java:88)
    at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:373)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:499)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:484)
    at jdk.internal.reflect.GeneratedMethodAccessor14.invoke(null:-1)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:568)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
    at java.lang.Thread.run(Thread.java:833)

```

Failed again today: https://gradle-enterprise.elastic.co/s/c2of3l5cjwn3q/


Reproduction line: 
```
./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=transform/transforms_start_stop/Test start/stop/start continuous transform}" -Dtests.seed=BEB94F78881131F1 -Dtests.locale=es-PY -Dtests.timezone=ACT -Druntime.java=17
```

Could not reproduce this locally.

```
java.lang.AssertionError: Failure at [transform/transforms_start_stop:251]: expected [2xx] status code but api [transform.stop_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:329)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:305)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1362)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1447)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1427)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:825)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"status_exception","reason":"Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [airline-transform-start-stop-continuous] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:329)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:305)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1362)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1447)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1427)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:825)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":409}]
        at __randomizedtesting.SeedInfo.seed([BEB94F78881131F1:36ED70A226ED5C09]:0)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:510)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:479)
        at jdk.internal.reflect.GeneratedMethodAccessor13.invoke(Unknown Source)
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:568)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
        at java.lang.Thread.run(Thread.java:833)
```

--- a/docs/changelog/90347.yaml
+++ b/docs/changelog/90347.yaml
@@ -1,0 +1,8 @@
+pr: 90347
+summary: Fix NPE in transform scheduling
+area: Transform
+type: bug
+issues:
+ - 88203
+ - 90301
+ - 90255

--- a/docs/changelog/90347.yaml
+++ b/docs/changelog/90347.yaml
@@ -3,6 +3,7 @@ summary: Fix NPE in transform scheduling
 area: Transform
 type: bug
 issues:
+ - 90356
  - 88203
  - 90301
  - 90255

--- a/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduledTask.java
+++ b/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduledTask.java
@@ -64,9 +64,7 @@ final class TransformScheduledTask {
             frequency,
             lastTriggeredTimeMillis,
             failureCount,
-            failureCount == 0
-                ? lastTriggeredTimeMillis + frequency.millis()
-                : calculateNextScheduledTimeAfterFailure(lastTriggeredTimeMillis, failureCount),
+            calculateNextScheduledTime(lastTriggeredTimeMillis, frequency, failureCount),
             listener
         );
     }
@@ -74,17 +72,24 @@ final class TransformScheduledTask {
     // Visible for testing
 
     /**
-     * Calculates the appropriate next scheduled time after a number of failures.
+     * Calculates the appropriate next scheduled time taking number of failures into account.
      * This method implements exponential backoff approach.
      *
      * @param lastTriggeredTimeMillis the last time (in millis) the task was triggered
+     * @param frequency               the frequency of the transform
      * @param failureCount            the number of failures that happened since the task was triggered
      * @return next scheduled time for a task
      */
-    static long calculateNextScheduledTimeAfterFailure(long lastTriggeredTimeMillis, int failureCount) {
+    static long calculateNextScheduledTime(Long lastTriggeredTimeMillis, TimeValue frequency, int failureCount) {
+        final long baseTime = lastTriggeredTimeMillis != null ? lastTriggeredTimeMillis : System.currentTimeMillis();
+
+        if (failureCount == 0) {
+            return baseTime + (frequency != null ? frequency : Transform.DEFAULT_TRANSFORM_FREQUENCY).millis();
+        }
+
         // Math.min(failureCount, 32) is applied in order to avoid overflow.
         long delayMillis = Math.min(Math.max((1L << Math.min(failureCount, 32)) * 1000, MIN_DELAY_MILLIS), MAX_DELAY_MILLIS);
-        return lastTriggeredTimeMillis + delayMillis;
+        return baseTime + delayMillis;
     }
 
     String getTransformId() {

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduledTaskTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/scheduling/TransformScheduledTaskTests.java
@@ -9,11 +9,13 @@ package org.elasticsearch.xpack.transform.transforms.scheduling;
 
 import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.transform.Transform;
 import org.elasticsearch.xpack.transform.transforms.scheduling.TransformScheduler.Listener;
 
 import java.time.Instant;
 
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.is;
 
 public class TransformScheduledTaskTests extends ESTestCase {
@@ -62,10 +64,10 @@ public class TransformScheduledTaskTests extends ESTestCase {
         }
     }
 
-    public void testCalculateNextScheduledTimeAfterFailure() {
+    public void testCalculateNextScheduledTimeExponentialBackoff() {
         long lastTriggeredTimeMillis = Instant.now().toEpochMilli();
         long[] expectedDelayMillis = {
-            5000,    // 5s
+            Transform.DEFAULT_TRANSFORM_FREQUENCY.millis(),    // normal schedule
             5000,    // 5s
             5000,    // 5s
             8000,    // 8s
@@ -85,9 +87,25 @@ public class TransformScheduledTaskTests extends ESTestCase {
         for (int failureCount = 0; failureCount < 1000; ++failureCount) {
             assertThat(
                 "failureCount = " + failureCount,
-                TransformScheduledTask.calculateNextScheduledTimeAfterFailure(lastTriggeredTimeMillis, failureCount),
+                TransformScheduledTask.calculateNextScheduledTime(lastTriggeredTimeMillis, null, failureCount),
                 is(equalTo(lastTriggeredTimeMillis + expectedDelayMillis[Math.min(failureCount, expectedDelayMillis.length - 1)]))
             );
         }
+    }
+
+    public void testCalculateNextScheduledTime() {
+        long now = Instant.now().toEpochMilli();
+        assertThat(
+            TransformScheduledTask.calculateNextScheduledTime(null, TimeValue.timeValueSeconds(10), 0),
+            is(greaterThanOrEqualTo(now + 10_000))
+        );
+        assertThat(
+            TransformScheduledTask.calculateNextScheduledTime(now, null, 0),
+            is(equalTo(now + Transform.DEFAULT_TRANSFORM_FREQUENCY.millis()))
+        );
+        assertThat(
+            TransformScheduledTask.calculateNextScheduledTime(null, null, 0),
+            is(greaterThanOrEqualTo(now + Transform.DEFAULT_TRANSFORM_FREQUENCY.millis()))
+        );
     }
 }

